tensorboard/plugins/debugger_v2/tf_debugger_v2_plugin/store/debugger_reducers_test.ts

```diff
@@ -253,71 +253,92 @@ describe('Debugger reducers', () => {
+            executions: createDebuggerExecutionsState({
+              displayCount,
+              scrollBeginIndex: 0,
+            }),
+          }); // `alerts` state is in an empty initial state.
+          const alert0 = createTestInfNanAlert({
+            op_type: 'RealDiv',
+            execution_index: firstAlertExecutionIndex,
+            graph_execution_trace_index: 100,
+          });
+          const alert1 = createTestInfNanAlert({
+            op_type: 'Log',
+            execution_index: firstAlertExecutionIndex + 1,
+            graph_execution_trace_index: firstAlertGraphExecutionIndex * 2,
+          });
+          const nextState = reducers(
+            state,
+            actions.alertsOfTypeLoaded({
+              numAlerts: 2,
+              alertsBreakdown: {
+                InfNanAlert: 2,
+              },
+              begin: 0,
+              end: 2,
+              alertType: 'InfNanAlert',
+              alerts: [alert0, alert1],
+            })
+          );
+          expect(nextState.alerts.alertsLoaded.state).toBe(
+            DataLoadState.LOADED
+          );
+          expect(
+            nextState.alerts.alertsLoaded.lastLoadedTimeInMs
+          ).toBeGreaterThan(0);
+          expect(nextState.alerts.numAlerts).toBe(2);
+          expect(nextState.alerts.alertsBreakdown).toEqual({
+            [AlertType.INF_NAN_ALERT]: 2,
+          });
+          expect(Object.keys(nextState.alerts.alerts)).toEqual([
+            AlertType.INF_NAN_ALERT,
+          ]);
+          const alertsOfType = nextState.alerts.alerts[AlertType.INF_NAN_ALERT];
+          expect(Object.keys(alertsOfType).length).toBe(2);
+          expect(alertsOfType[0]).toEqual(alert0);
+          expect(alertsOfType[1]).toEqual(alert1);
+          expect(Object.keys(nextState.alerts.executionIndices)).toEqual([
+            AlertType.INF_NAN_ALERT,
+          ]);
+          const executionIndices =
+            nextState.alerts.executionIndices[AlertType.INF_NAN_ALERT];
+          expect(executionIndices).toEqual([
+            firstAlertExecutionIndex,
+            firstAlertExecutionIndex + 1,
+          ]);
+          const graphExecutionIndices =
+            nextState.alerts.graphExecutionIndices[AlertType.INF_NAN_ALERT];
+          expect(graphExecutionIndices).toEqual([
+            firstAlertGraphExecutionIndex,
+            firstAlertGraphExecutionIndex * 2,
+          ]);
+          // Verify that the top-level execution for the first alert is scrolled
+          // into view.
+          expect(nextState.executions.scrollBeginIndex).toBe(
+            expectedScrollBegin
+          );
+          // Verify that the intra-graph execution for the first alert with a
+          // graph_execution_trace_index is focused on.
+          expect(nextState.graphExecutions.focusIndex).toBe(
+            firstAlertGraphExecutionIndex
+          );
+        }
+      );
     }
 
     it('Updates alerts data: non-empty initial state', () => {
```

Comment:
Is there truly nothing interesting here? How about `alert.graph_execution_trace_index` is null and it is set to out of bound on the new alert?

Reply:
Good question. Added some assertions in this test method for the new logic related to `graphExecutionIndices`.